handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -135,45 +131,97 @@ private void checkTrusted(String type, X509Certificate[] chain) throws Certifica
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(Iterable<String> fingerprints) {
-        this(toFingerprintArray(fingerprints));
+        this("SHA1", toFingerprintArray(fingerprints));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints in hexadecimal form
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(String... fingerprints) {
-        this(toFingerprintArray(Arrays.asList(fingerprints)));
+        this("SHA1", toFingerprintArray(Arrays.asList(fingerprints)));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(byte[]... fingerprints) {
+        this("SHA1", fingerprints);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param algorithm a hash algorithm
+     * @param fingerprints a list of fingerprints
+     */
+    private FingerprintTrustManagerFactory(final String algorithm, byte[][] fingerprints) {
+        ObjectUtil.checkNotNull(algorithm, "algorithm");
         ObjectUtil.checkNotNull(fingerprints, "fingerprints");
 
+        this.tlmd = new FastThreadLocal<MessageDigest>() {
+
+            @Override
+            protected MessageDigest initialValue() {
+                try {
+                    return MessageDigest.getInstance(algorithm);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalArgumentException("Unsupported hash algorithm", e);
+                }
+            }
+        };
+
+        int hashLength = tlmd.get().getDigestLength();
         List<byte[]> list = new ArrayList<byte[]>(fingerprints.length);
         for (byte[] f: fingerprints) {
             if (f == null) {
                 break;
             }
-            if (f.length != SHA1_BYTE_LEN) {
-                throw new IllegalArgumentException("malformed fingerprint: " +
-                        ByteBufUtil.hexDump(Unpooled.wrappedBuffer(f)) + " (expected: SHA1)");
+            if (f.length != hashLength) {
+                throw new IllegalArgumentException(
+                        "malformed fingerprint (length): " + ByteBufUtil.hexDump(Unpooled.wrappedBuffer(f)));
             }
             list.add(f.clone());
         }
 
         this.fingerprints = list.toArray(new byte[0][]);
```

Comment:
`EmptyArrays.BYTES`

Reply:
If I am not missing something, that could not be used here because `EmptyArrays.EMPTY_BYTES` is `byte[]`.

https://github.com/netty/netty/blob/00afb19d7a37de21b35ce4f6cb3fa7f74809f2ab/common/src/main/java/io/netty/util/internal/EmptyArrays.java#L28

Reply:
ah sorry I missed that it is `[][]`